core/operators.h

```diff
@@ -48,6 +48,10 @@ struct times_equals {
       pair_self_t<double, float, int32_t, int64_t>{},
       pair_custom_t<std::pair<double, float>, std::pair<float, double>,
                     std::pair<int64_t, int32_t>,
+                    std::pair<double, Bool>,
+                    std::pair<float, Bool>,
+                    std::pair<int64_t, Bool>,
+                    std::pair<int32_t, Bool>,
```

Comment:
Shouldn't this be `bool`? `Bool` should never be used anywhere outside the actual implementation of details of `Variable`.